### PR TITLE
Add support for creating and deleting Route53 hosted zones

### PR DESCRIPTION
**Pull request by minichate**
Supports both private (per-VPC) and public hosted zones.

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks for submitting this module to Ansible Extras.

To help facilitate reviews, we’ve broadened the number of people who can approve modules for inclusion into the Extras repository. The list of official reviewers can be found here:

https://github.com/ansible/ansible-modules-extras/blob/devel/REVIEWERS.md 

Our new policy is that if a new module is reviewed and approved by at least two official module reviewers, the module will be approved for inclusion.  We will be asking the community of reviewers to take a look at these modules on a regular basis.

To ensure that your module has the best chance of being approved, please double-check that you adhere to the Ansible module guidelines: http://docs.ansible.com/developing_modules.html#module-checklist

**Comment by gregdek** [comment]
Oh, also, we have new guidelines for AWS modules, so you might want to look those over as well and make sure that this module adheres to those standards: https://github.com/ansible/ansible-modules-extras/blob/devel/cloud/amazon/GUIDELINES.md

File: cloud/amazon/route53_zone.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
+#!/usr/bin/python
+
+DOCUMENTATION = '''
+module: route53_zone
+short_description: add or delete Route53 zones
```
Comment (bcoca) [inline cloud/amazon/route53_zone.py:19]: ~~needs version_added~~ fixed

File: cloud/amazon/route53_zone.py [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+description:
+    - Creates and deletes Route53 private and public zones
+options:
+    zone:
+        description:
+            - The DNS zone record (eg: foo.com.)
+        required: true
+        default: null
```
Comment (bcoca) [inline cloud/amazon/route53_zone.py]: no need for default when required is true

File: cloud/amazon/route53_zone.py [hunk 3]
```diff
@@ -0,0 +1,148 @@
+#!/usr/bin/python
+
+DOCUMENTATION = '''
+module: route53_zone
```
Comment (bcoca) [inline cloud/amazon/route53_zone.py:18]: ~~author is missing~~ fixed

File: cloud/amazon/route53_zone.py [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+short_description: add or delete Route53 zones
+description:
+    - Creates and deletes Route53 private and public zones
+options:
+    zone:
+        description:
+            - The DNS zone record (eg: foo.com.)
+        required: true
+        default: null
+    command:
```
Comment (bcoca) [inline cloud/amazon/route53_zone.py]: instead of command we would prefer the use of state=absent|present

Comment (bcoca) [inline cloud/amazon/route53_zone.py]: also document the choices here

Comment (minichate) [inline cloud/amazon/route53_zone.py]: Will do! I stole the `command` to match from http://docs.ansible.com/route53_module.html. But I can easily change it to `state` :+1:

Comment (bcoca) [inline cloud/amazon/route53_zone.py]: we do have a few modules that departed from the pattern in the past, confused a lot of people, that is why we are being stricter about it now

**Comment by minichate** [comment]
I believe the Travis CI build failure isn't my fault -- I'll close/reopen the PR to trigger a new build

**Comment by minichate** [comment]
Build failure appears to be in `monitoring/sensu_check.py` which is outside of this PR.

**Comment by minichate** [comment]
(Ahh yes: it's since https://github.com/ansible/ansible-modules-extras/pull/203 merged with python2.4 failures)

**Comment by bcoca** [comment]
rebase to latest devel, I fixed the sensu issue, but you don't have that commit (you might need to force push on this branch).

**Comment by minichate** [comment]
Rebase fixed the issue! :+1:

**Comment by gregdek** [comment]
Thx @minichate. @bcoca can you re-review?

**Comment by minichate** [comment]
Thanks folks! Let me know if you'd like me to change anything, @bcoca

File: cloud/amazon/route53_zone.py [hunk 5]
```diff
@@ -0,0 +1,150 @@
+#!/usr/bin/python
+
```
Comment (bcoca) [inline cloud/amazon/route53_zone.py:16]: license is missing

**Comment by bcoca** [comment]
just add GPL license header as all modules should have, then we are good to merge

**Comment by minichate** [comment]
Hey @bcoca, figured I'd ping you to see if you're ok with the that of this PR as-is?

**Pull request merged**